src/GraphQL/Resolvers/NameFieldResolver.cs

```diff
@@ -60,7 +60,9 @@ public class NameFieldResolver : IFieldResolver
         /// <param name="target"> The type from which you want to get the value. </param>
         /// <param name="name"> Property/method name. </param>
         /// <returns> Compiled field resolver. </returns>
-        private static IFieldResolver CreateResolver(Type target, string name)
+        private static IFieldResolver CreateResolver(
```

Comment:
I might just mark this entire class with `[RequiresUnreferencedCode]` as the use of `NameFieldResolver` is not supported (although if the type was rooted, then it should work fine).  Marking the entire class will eliminate all warnings here and we can revert this line of code.

Reply:
Again, we will need to ignore warnings from callers.  (or else we will end up marking the entirety of GraphQL.NET with `RequiresUnreferencedCode` !)

Reply:
> Again, we will need to ignore warnings from callers.

I don't understand why so. If we suppress warnings, then this is provided by some guarantees in the code.